### PR TITLE
Refactor, minor simplify to highlighted units collection usage

### PR DESCRIPTION
**Pull request by DanVanAtta**

### CODE REVIEW

File: game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java [hunk 1]
```diff
@@ -618,11 +618,10 @@ public void mouseEnter(final List<Unit> units, final Territory territory) {
           final boolean isCorrectTerritory =
               (firstSelectedTerritory == null) || firstSelectedTerritory.equals(territory);
           if (someOwned && isCorrectTerritory) {
-            final Collection<List<Unit>> highlight = new ArrayList<>();
-            highlight.add(units);
-            getMap().setUnitHighlight(highlight);
+            getMap()
+                .setUnitHighlight(Collections.unmodifiableList(Collections.singletonList(units)));
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java]: I believe the order of those should be swapped

File: game-core/src/main/java/games/strategy/triplea/ui/unit/scroller/UnitScroller.java [hunk 2]
```diff
@@ -247,9 +246,7 @@ private void centerOnMovableUnit(final boolean selectNext) {
       if (!matchedUnits.isEmpty()) {
         drawUnitAvatarPane(t);
         newFocusedTerritory = t;
-        final Collection<List<Unit>> highlight = new ArrayList<>();
-        highlight.add(matchedUnits);
-        mapPanel.setUnitHighlight(highlight);
+        mapPanel.setUnitHighlight(Collections.singleton(matchedUnits));
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/triplea/ui/unit/scroller/UnitScroller.java]: Is matched units guaranteed to be immutable?

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/triplea/ui/unit/scroller/UnitScroller.java]: Yes, `matched` is local to the for loop, there is a `break` that comes next exiting the for loop.

File: game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java]: updated: 0465ba106

**Pull request merged**